Content.Server/Power/Pow3r/BatteryRampPegSolver.cs

```diff
@@ -115,11 +115,10 @@ private void UpdateNetwork(Network network, PowerState state, float frameTime)
 
                 var batterySpace = (battery.Capacity - battery.CurrentStorage) * (1 / battery.Efficiency);
                 batterySpace = Math.Max(0, batterySpace);
-                var scaledSpace = batterySpace / frameTime;
 
-                var chargeRate = battery.MaxChargeRate + battery.LoadingNetworkDemand / battery.Efficiency;
+                var chargeRate = Math.Min(battery.MaxChargeRate, batterySpace) + battery.LoadingNetworkDemand / battery.Efficiency;
```

Comment:
The units in this change do not make sense.

- `battery.MaxChargeRate` is a rate of change of energy (not charge), i.e. power in Watts (**Joules/sec**).
- `batterySpace` is in terms of capacity (**Joules**).
- This is taking the `Math.Min` of two quantities with incompatible units.

For all intents and purposes, `frameTime` is a constant, so `scaledSpace` is used to scale the battery's desired power proportional to the amount of space left in the battery. In other words, a battery that is mostly empty will have more desired power than a battery that is almost full.